cachito/workers/pkg_managers/gomod.py

```diff
@@ -191,50 +191,62 @@ def resolve_gomod(app_source_path, request, dep_replacements=None):
             shutil.copytree(tmp_download_cache_dir, str(bundle_dir.gomod_download_dir))
 
         log.info("Retrieving the list of package level dependencies")
-        go_list_deps_output = run_gomod_cmd(
-            ("go", "list", "-deps", "-f", "{{if not .Standard}}{{.ImportPath}} {{.Module}}{{end}}"),
-            run_params,
-        )
-
-        pkg_level_deps = []
-        for line in go_list_deps_output.splitlines():
-            # line example: pkg.io/foo/bar pkg.io/foo/var v1.0.0 => pkg.io/foo/bar v1.0.1
-            parts = [part for part in line.split(" ")]
+        list_pkgs_cmd = ("go", "list", "-find", "./...")
+        go_list_pkgs_output = run_gomod_cmd(list_pkgs_cmd, run_params)
+        packages = []
+        for package in go_list_pkgs_output.splitlines():
+            if next((d for p in packages for d in p["pkg_deps"] if d["name"] == package), False):
```

Comment:
This is hard to read and iterates through the same data repeatedly. How about a set called `processed_pkg_deps` and then you can change this line to the following?
```suggestion
            if package in processed_pkg_deps:
```

It does use more memory, but then for every package in your iteration, it's only an O(1) operation and it's easier to read.